relayer/morse/protocol.go

```diff
@@ -280,38 +250,28 @@ func (p *Protocol) fetchSessions() map[string]provider.Session {
 	return sessions
 }
 
-func (p *Protocol) sendRelay(
-	chainID string,
-	node provider.Node,
-	session provider.Session,
-	aat provider.PocketAAT,
-	timeoutMillisec int,
-	payload relayer.Payload,
-) (provider.RelayOutput, error) {
-	fullNodeInput := &sdkrelayer.Input{
-		Blockchain: chainID,
-		Node:       &node,
-		Session:    &session,
-		PocketAAT:  &aat,
-		Data:       payload.Data,
-		Method:     payload.Method,
-		Path:       payload.Path,
-	}
-
-	timeout := timeoutMillisec
-	if timeout == 0 {
-		timeout = defaultRelayTimeoutMillisec
-	}
-	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout)*time.Millisecond)
-	defer cancel()
+// TODO_UPNEXT(@adshmh): Refactor all caching out of the Protocol struct, and use an interface to access Apps and Sessions, and send relays.
+// Then add 2 implementations of the FullNode interface:
+// CachingFullNode
+// LazyFullNode
+//
+// getAppsUniqueEndpoints returns a map of all endpoints matching the provided service ID.
+// If an endpoint matches a service ID through multiple apps/sessions, only a single entry
```

Comment:
I think this could happen because there is no guarantee on sessions having unique nodes/endpoints.
e.g. if there are only 30 Morse endpoints stakes for some service, there will be some overlap of endpoints between two sessions corresponding to two different applications.